ipn/ipnlocal/local.go

```diff
@@ -3460,8 +3460,11 @@ func (b *LocalBackend) reconfigAppConnectorLocked(nm *netmap.NetworkMap, prefs i
 	slices.SortFunc(routes, func(i, j netip.Prefix) int { return i.Addr().Compare(j.Addr()) })
 	domains = slices.Compact(domains)
 	routes = slices.Compact(routes)
-	b.appConnector.UpdateRoutes(routes)
-	b.appConnector.UpdateDomains(domains)
+	// The following updates may want to take preferences changes, so make them concurrent.
+	go func() {
+		b.appConnector.UpdateRoutes(routes)
+		b.appConnector.UpdateDomains(domains)
+	}()
```

Comment:
do you care about scheduling order?

Reply:
(Maybe we should move `type execQueue` out somewhere shared... that type runs func(s) in another goroutine but in order....)

Reply:
yeah, I think you're right